AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt

```diff
@@ -0,0 +1,702 @@
+
+    LaunchedEffect(Unit) {
+        loadWeather()
+    }
+    val lazyListState = rememberLazyListState()
+
+    // The background color. The value is changed by the current tab.
+    val backgroundColor by animateColorAsState(if (currentTab == 0) purple100 else green300)
+
+    // The coroutine scope for event handlers calling suspend functions.
+    val scope = rememberCoroutineScope()
+    Scaffold(
+        topBar = {
+            HomeTabBar(
+                backgroundColor = backgroundColor,
+                currentTab = currentTab,
+                onTabSelected = { currentTab = it }
+            )
+        },
+        backgroundColor = backgroundColor,
+        floatingActionButton = {
+            HomeFloatingActionButton(
+                extended = lazyListState.isScrollingUp(),
+                onClick = {
+                    scope.launch {
+                        showEditMessage()
+                    }
+                }
+            )
+        }
+    ) {
+        LazyColumn(
+            contentPadding = PaddingValues(start = 16.dp, top = 32.dp, end = 16.dp, bottom = 32.dp),
+            state = lazyListState
+        ) {
+            // Weather
+            item { Header(title = stringResource(R.string.weather)) }
+            item { Spacer(modifier = Modifier.height(16.dp)) }
+            item {
+                Surface(
+                    modifier = Modifier.fillMaxWidth(),
+                    elevation = 2.dp
+                ) {
+                    if (weatherLoading) {
+                        LoadingRow()
+                    } else {
+                        WeatherRow(onRefresh = {
+                            scope.launch {
+                                loadWeather()
+                            }
+                        })
+                    }
+                }
+            }
+            item { Spacer(modifier = Modifier.height(32.dp)) }
+
+            // Topics
+            item { Header(title = stringResource(R.string.topics)) }
+            item { Spacer(modifier = Modifier.height(16.dp)) }
+            items(count = allTopics.size) { i ->
+                val topic = allTopics[i]
+                TopicRow(
+                    topic = topic,
+                    expanded = expandedTopic == topic,
+                    onClick = {
+                        expandedTopic = if (expandedTopic == topic) null else topic
+                    }
+                )
+            }
+            item { Spacer(modifier = Modifier.height(32.dp)) }
+
+            // Tasks
+            item { Header(title = stringResource(R.string.tasks)) }
+            item { Spacer(modifier = Modifier.height(16.dp)) }
+            if (tasks.isEmpty()) {
+                item {
+                    TextButton(onClick = { tasks.clear(); tasks.addAll(allTasks) }) {
+                        Text(stringResource(R.string.add_tasks))
+                    }
+                }
+            }
+            items(count = tasks.size) { i ->
```

Comment:
`items(items = tasks, key = { it }) ( task ->`

Reply:
Is a key necessary here, and if so, why?

Reply:
I filed an issue: b/180864167